src/common/src/buffer/bitmap.rs

```diff
@@ -461,12 +470,51 @@ mod tests {
 
     #[test]
     fn test_bitwise_and() {
-        let bitmap1 = Bitmap::from_bytes(Bytes::from_static(&[0b01101010]));
-        let bitmap2 = Bitmap::from_bytes(Bytes::from_static(&[0b01001110]));
-        assert_eq!(
-            Bitmap::from_bytes(Bytes::from_static(&[0b01001010])),
-            (&bitmap1 & &bitmap2)
-        );
+        #[rustfmt::skip]
+        let cases = [(
+            vec![],
+            vec![],
+            vec![],
+        ), (
+            vec![0, 1, 1, 0, 1, 0, 1, 0],
+            vec![0, 1, 0, 0, 1, 1, 1, 0],
+            vec![0, 1, 0, 0, 1, 0, 1, 0],
+        ), (
+            vec![0, 1, 0, 1, 0],
+            vec![1, 1, 1, 1, 0],
+            vec![0, 1, 0, 1, 0],
+        )];
+
+        for (input1, input2, expected) in cases {
+            let bitmap1: Bitmap = input1.into_iter().map(|x| x != 0).collect();
+            let bitmap2: Bitmap = input2.into_iter().map(|x| x != 0).collect();
+            let res = &bitmap1 & &bitmap2;
+            assert_eq!(res.iter().map(|x| x as i32).collect::<Vec<_>>(), expected,);
+        }
+    }
+
+    #[test]
+    fn test_bitwise_not() {
+        #[rustfmt::skip]
+        let cases = [(
+            vec![1, 0, 1, 0, 1],
+            vec![0, 1, 0, 1, 0],
+        ), (
+            vec![],
+            vec![],
+        ), (
+            vec![1, 0, 1, 1, 0, 0, 1, 1],
+            vec![0, 1, 0, 0, 1, 1, 0, 0],
+        ), (
+            vec![1, 0, 0, 1, 1, 1, 0, 0, 1, 0],
+            vec![0, 1, 1, 0, 0, 0, 1, 1, 0, 1],
+        )];
+
+        for (input, expected) in cases {
+            let bitmap: Bitmap = input.into_iter().map(|x| x != 0).collect();
+            let res = !&bitmap;
+            assert_eq!(res.iter().map(|x| x as i32).collect::<Vec<_>>(), expected);
```

Comment:
The test passed because `num_bits` is correct, so `iter` is correct. `num_high_bits` is wrong.

Reply:
Fully understood! Will update the tests